UnitsNet.Tests/QuantityTypeConverterTest.cs

```diff
@@ -0,0 +1,348 @@
+
+            Assert.True(unitsNETTypeConverter.CanConvertTo(value));
+        }
+
+        [Theory]
+        [InlineData(typeof(double))]
+        [InlineData(typeof(object))]
+        [InlineData(typeof(float))]
+        [InlineData(typeof(Length))]
+        public void CanConvertTo_GiveSomeTypes_ReturnsFalse(Type value)
+        {
+            UnitsNetTypeConverter<Length> unitsNETTypeConverter = new UnitsNetTypeConverter<Length>();
+
+            Assert.False(unitsNETTypeConverter.CanConvertTo(value));
+        }
+
+        [Fact]
+        public void ConvertFrom_GiveSomeStrings()
+        {
+            UnitsNetTypeConverter<Length> unitsNETTypeConverter = new UnitsNetTypeConverter<Length>();
+            ITypeDescriptorContext context = new TypeDescriptorContext("SomeMemberName", new Attribute[] { });
+
+            Assert.Equal(Length.FromMillimeters(1), unitsNETTypeConverter.ConvertFrom(context, culture, "1mm"));
+            Assert.Equal(Length.FromMeters(1), unitsNETTypeConverter.ConvertFrom(context, culture, "1m"));
+            Assert.Equal(Length.FromMeters(1), unitsNETTypeConverter.ConvertFrom(context, culture, "1"));
+            Assert.Equal(Length.FromKilometers(1), unitsNETTypeConverter.ConvertFrom(context, culture, "1km"));
+
+            context = new TypeDescriptorContext("SomeMemberName", new Attribute[] { new DefaultUnitAttribute(Units.LengthUnit.Centimeter) });
+
+            Assert.Equal(Length.FromMillimeters(1), unitsNETTypeConverter.ConvertFrom(context, culture, "1mm"));
+            Assert.Equal(Length.FromMeters(1), unitsNETTypeConverter.ConvertFrom(context, culture, "1m"));
+            Assert.Equal(Length.FromCentimeters(1), unitsNETTypeConverter.ConvertFrom(context, culture, "1"));
+            Assert.Equal(Length.FromKilometers(1), unitsNETTypeConverter.ConvertFrom(context, culture, "1km"));
+
+            context = new TypeDescriptorContext("SomeMemberName", new Attribute[]
+                {
+                    new DefaultUnitAttribute(Units.LengthUnit.Centimeter),
+                    new ConvertToUnitAttribute(Units.LengthUnit.Meter)
+                });
+
+            Assert.Equal(Length.FromMeters(0.001), unitsNETTypeConverter.ConvertFrom(context, culture, "1mm"));
+            Assert.Equal(Length.FromMeters(1), unitsNETTypeConverter.ConvertFrom(context, culture, "1m"));
+            Assert.Equal(Length.FromMeters(0.01), unitsNETTypeConverter.ConvertFrom(context, culture, "1"));
+            Assert.Equal(Length.FromMeters(1000), unitsNETTypeConverter.ConvertFrom(context, culture, "1km"));
+        }
+
+        [Fact]
+        public void ConvertFrom_GiveEmpyString()
+        {
+            UnitsNetTypeConverter<Length> unitsNETTypeConverter = new UnitsNetTypeConverter<Length>();
+            ITypeDescriptorContext context = new TypeDescriptorContext("SomeMemberName", new Attribute[] { });
+
+            Assert.Throws<NotSupportedException>(() => unitsNETTypeConverter.ConvertFrom(context, culture, ""));
```

Comment:
Is `NotSupportedException` the norm for type converters receiving invalid/unsupported input? Just asking, I don't know. I see you get ArgumentException below for unrecognized quantity strings, so I would maybe expect empty string to also throw ArgumentException.

Reply:
I tried to follow the schema Microsoft is using.
https://docs.microsoft.com/en-us/dotnet/api/system.componentmodel.uint16converter?view=netframework-4.8

Reply:
Thanks for the reference! It's good then.